js/angular-edrna/SubConverter.js

```diff
@@ -0,0 +1,373 @@
+			// lines = lines.filter(function(line) { return line.match(tagRegex) !== null; });
+
+			lines.forEach(function(line) {
+				var matched = line.match(tagRegex);
+
+				if(matched === null) return;
+
+				var start = parseFloat(matched[1], 10) * 1000;
+				var end = parseFloat(matched[4], 10) * 1000;
+				var text = line.replace(tagRegex, '');
+				var el = angular.element(document.createElement('textarea'));
+				var epoch = new Date(0);
+				epoch.setHours(0);
+				start = new Date(epoch.getTime() + start);
+				end = new Date(start.getTime() + end);
+				text = text.replace(/\&amp;/g, '&');
+				text = el.html(text).text();
+
+				parsed.push({
+					start: {
+						h: start.getHours(),
+						m: start.getMinutes(),
+						s: start.getSeconds(),
+						mili: start.getMilliseconds()
+					},
+					end: {
+						h: end.getHours(),
+						m: end.getMinutes(),
+						s: end.getSeconds(),
+						mili: end.getMilliseconds()
+					},
+					data: text.split('\n')
+				});
+			});
+
+
+			return parsed;
+		});
+
+		SubConverterProvider.addDstConverter('text/x-ssa', function(data) {
+			var result = '';
+			result += '[Script Info]\n';
+			result += '; Script generated by Junction\n';
+			result += '; http://www.junctioneducation.com/\n';
+			result += 'ScriptType: v4.00+\n';
+			result += 'WrapStyle: 0\n';
+			result += 'PlayResX: 640\n';
+			result += 'PlayResY: 480\n';
+			result += 'ScaledBorderAndShadow: yes\n';
+			result += 'VideoAspectRatio: 0\n';
+			result += 'VideoZoom: 6\n';
+			result += 'VideoPosition: 0\n';
+			result += 'Last Style Storage: Default\n';
+			result += '\n';
+			result += '[V4+ Styles]\n';
+			result += 'Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, Encoding\n';
+			result += 'Style: Default,Arial,25,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,3,1,1,2,10,10,30,1\n';
+			result += '\n';
+			result += '[Events]\n';
+			result += 'Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, Effect, Text\n';
+
+			function toFixed(number, len) {
+				while((number + '').length > len) {
+					number = (number + '').substring(0, (number + '').length-1);
+				}
+
+				while((number + '').length < len) {
+					number = '0' + number;
+				}
+
+				return number;
+			}
+
+			data.forEach(function(line) {
+				result += 'Dialogue: ';
+				result += '0,';
+				result += toFixed(line.start.h, 1) + ':' + toFixed(line.start.m, 2) + ':' + toFixed(line.start.s, 2) + '.' + toFixed(line.start.mili, 2) + ',';
+				result += toFixed(line.end.h, 1) + ':' + toFixed(line.end.m, 2) + ':' + toFixed(line.end.s, 2) + '.' + toFixed(line.end.mili, 2) + ',';
+				result += 'Default,,0,0,0,,';
+				result += line.data.join('\\N');
+				result += '\n';
+			});
+			
+			return result;
+		});
+	}
+})();
```

Comment:
I don't see any usage example. I think that would be great to see how people can use it.